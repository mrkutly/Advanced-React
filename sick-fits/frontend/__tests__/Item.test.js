import Item from "../components/Item";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

const fakeItem = {
	id: "ABC123",
	title: "A cool item",
	price: 1500,
	description: "This item rips!",
	image: "dog.jpg",
	largeImage: "large-dog.jpg",
};

describe("<Item />", () => {
	it("renders and matches the snapshot", () => {
		const wrapper = shallow(<Item item={fakeItem} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	it("renders and displays the price and title", () => {
		const wrapper = shallow(<Item item={fakeItem} />);
		const PriceTag = wrapper.find("PriceTag");

		// ? .dive() shallow renders a component
		// ? this allows you to shallow render one level deeper for a specific
		// ? child of the component that you are shallow rendering
		expect(PriceTag.dive().text()).toBe("$15");

		// ? .find() can take in multiple selectors to allow you to traverse your
		// ? component tree
		expect(wrapper.find("Title a").text()).toBe(fakeItem.title);
	});

	it("renders an image", () => {
		const wrapper = shallow(<Item item={fakeItem} />);
		const image = wrapper.find("img");

		expect(image.props().src).toBe(fakeItem.image);
		expect(image.props().alt).toBe(fakeItem.title);
	});

	it("renders the buttons", () => {
		const wrapper = shallow(<Item item={fakeItem} />);
		const buttonList = wrapper.find(".buttonList");

		expect(buttonList.children()).toHaveLength(3);
		expect(buttonList.find("Link")).toHaveLength(1);
		expect(buttonList.find("DeleteItem")).toHaveLength(1);
		expect(buttonList.find("AddToCart")).toHaveLength(1);
	});
});
