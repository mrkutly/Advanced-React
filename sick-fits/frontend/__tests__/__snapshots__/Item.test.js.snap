// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $15
  </PriceTag>
  <p>
    This item rips!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      item={
        Object {
          "description": "This item rips!",
          "id": "ABC123",
          "image": "dog.jpg",
          "largeImage": "large-dog.jpg",
          "price": 1500,
          "title": "A cool item",
        }
      }
    />
    <DeleteItem
      id="ABC123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
